src/core/expr/cumsum.cc

```diff
@@ -0,0 +1,125 @@
+#include "expr/fexpr_func.h"
+#include "column/virtual.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "stype.h"
+#include "python/xargs.h"
+#include "documentation.h"
+#include <numeric>
+#include <vector>
+
+namespace dt {
+namespace expr {
+
+template <typename T>
+class Column_cumsum : public Virtual_ColumnImpl{
+    private:
+      Column acol_;
+
+    public:
+      Column_cumsum(Column&& a)
+        : Virtual_ColumnImpl(a.nrows(), a.stype()),
+          acol_(std::move(a))
+
+    {
+      xassert(acol_.can_be_read_as<T>());
+    }
+
+      ColumnImpl* clone() const override {
+          return new Column_cumsum(Column(acol_));
+      }
+
+      size_t n_children() const noexcept override {
+      return 1;
+    }
+
+    const Column& child(size_t) const override {
+      return acol_;
+    }
+
+
+    bool get_element(size_t i, T* out) const override {
```

Comment:
@samukweku `i` is the row id that is passed to the `get_element()` method. `j` goes from `0` to `i` and `tmp` accumulates that sum of elements from `0`th to `i`th.